src/test/java/teammates/sqlui/webapi/QueryLogsActionTest.java

```diff
@@ -0,0 +1,404 @@
+package teammates.sqlui.webapi;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.logs.EmailSentLogDetails;
+import teammates.common.datatransfer.logs.ExceptionLogDetails;
+import teammates.common.datatransfer.logs.FeedbackSessionAuditLogDetails;
+import teammates.common.datatransfer.logs.GeneralLogEntry;
+import teammates.common.datatransfer.logs.LogSeverity;
+import teammates.common.datatransfer.logs.RequestLogDetails;
+import teammates.common.datatransfer.logs.RequestLogUser;
+import teammates.common.datatransfer.logs.SourceLocation;
+import teammates.common.util.Const;
+import teammates.ui.output.GeneralLogsData;
+import teammates.ui.webapi.JsonResult;
+import teammates.ui.webapi.QueryLogsAction;
+
+/**
+ * SUT: {@link QueryLogsAction}.
+ */
+public class QueryLogsActionTest extends BaseActionTest<QueryLogsAction> {
+    private static final String GOOGLE_ID = "user-googleId";
+
+    JsonResult actionOutput;
```

Comment:
I think this can be local variables inside the test methods, rather than an instance variable, since it should not be shared across the test cases